src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawerCreateForm.tsx

```diff
@@ -40,161 +41,168 @@ type TemplatesCatalogDrawerCreateFormProps = {
 
-      const [vmName, setVMName] = React.useState(initialVMName || '');
-      const [startVM, setStartVM] = React.useState(true);
-      const [isQuickCreating, setIsQuickCreating] = React.useState(false);
-      const [quickCreateError, setQuickCreateError] = React.useState(undefined);
-      const [models, modelsLoading] = useK8sModels();
+    const [vmName, setVMName] = useState(initialVMName || '');
+    const [startVM, setStartVM] = useState(true);
+    const [isQuickCreating, setIsQuickCreating] = useState(false);
+    const [quickCreateError, setQuickCreateError] = useState(undefined);
+    const [models, modelsLoading] = useK8sModels();
 
-      const [processedTemplateAccessReview] = useAccessReview({
-        namespace,
-        resource: ProcessedTemplatesModel.plural,
-        verb: 'create',
-      });
+    const [processedTemplateAccessReview] = useAccessReview({
+      namespace,
+      resource: ProcessedTemplatesModel.plural,
+      verb: 'create',
+    });
 
-      const onQuickCreate = () => {
-        setIsQuickCreating(true);
-        setQuickCreateError(undefined);
+    const onQuickCreate = () => {
+      setIsQuickCreating(true);
+      setQuickCreateError(undefined);
 
-        const parameterForName = extractParameterNameFromMetadataName(template) || 'NAME';
+      const parameterForName = extractParameterNameFromMetadataName(template) || 'NAME';
 
-        const templateToProcess = produce(template, (draftTemplate) => {
-          if (parameterForName)
-            replaceTemplateParameterValue(draftTemplate, parameterForName, vmName);
-        });
+      const templateToProcess = produce(template, (draftTemplate) => {
+        if (parameterForName)
+          replaceTemplateParameterValue(draftTemplate, parameterForName, vmName);
+      });
 
-        quickCreateVM({
-          template: templateToProcess,
-          models,
-          overrides: { name: vmName, namespace, startVM },
+      quickCreateVM({
+        template: templateToProcess,
+        models,
+        overrides: { name: vmName, namespace, startVM },
+      })
+        .then((vm) => {
+          setIsQuickCreating(false);
+          history.push(getResourceUrl({ model: VirtualMachineModel, resource: vm }));
         })
-          .then((vm) => {
-            setIsQuickCreating(false);
-            history.push(getResourceUrl({ model: VirtualMachineModel, resource: vm }));
-          })
-          .catch((err) => {
-            setIsQuickCreating(false);
-            setQuickCreateError(err);
-          });
-      };
+        .catch((err) => {
+          setIsQuickCreating(false);
+          setQuickCreateError(err);
+        });
+    };
 
-      const onCustomize = (e) => {
-        e.preventDefault();
-        let catalogUrl = `templatescatalog/customize?name=${template.metadata.name}&namespace=${template.metadata.namespace}&defaultSourceExists=${canQuickCreate}`;
+    const onCustomize = (e: { preventDefault: () => void }) => {
```

Comment:
the right type is event, there is many events, u just need to find the right one, most of the time typescript will let u know when you highlight where you are using it